chat.delta.desktop.yml

```diff
@@ -0,0 +1,120 @@
+
+
+modules:
+  - name: cyrus-sasl2
+    # Weird that we can disable it; I thought it was a dependency. Hm.
+    disabled: true
+    sources:
+        - type: archive
+          url: https://www.cyrusimap.org/releases/cyrus-sasl-2.1.27.tar.gz
+          sha256: 26866b1549b00ffd020f188a43c258017fa1c382b3ddadd8201536f72efb05d5
+
+
+  - name: nodejs
+    cleanup:
+        - "/include"
+        - "/share"
+        - "/app/lib/node_modules/npm/changelogs"
+        - "/app/lib/node_modules/npm/doc"
+        - "/app/lib/node_modules/npm/html"
+        - "/app/lib/node_modules/npm/man"
+        - "/app/lib/node_modules/npm/scripts"       
+    sources:
+        - type: archive
+          url: https://nodejs.org/dist/v8.13.0/node-v8.13.0.tar.xz
+          sha256: 2aa99474a336c6339d14f08cc27d387c5168e6fb6cbcaaea1d5ff7aa89642de2
+  
+  - name: delta
+    buildsystem: simple
+    build-options:
+        env:
+            electron_config_cache: /run/build/delta/npm-cache
+            npm_config_nodedir: /app/
+            #npm_config_devdir: /run/build/delta/npm-dev
+            npm_config_devdir: /app
+            ELECTRON_CACHE: /run/build/delta/npm-cache
+    build-commands:
+        - npm install  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        - npm run build  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        # This seems to be the appropriate command equivalent to a "make install".
+        # - ./node_modules/.bin/electron-builder build --linux dir
+        # It seems to be provided by the "pack" shortcut.
+        - npm run pack  --offline --cache=/run/build/delta/npm-cache/  --verbose
+        - mkdir -p /app/delta /app/bin
+        - cp -ar dist/linux-unpacked/* /app/delta/
+        # Not knowing how else to install the app into a certain prefix, we copy everything :(
+        # - cp -ra * /app/
+        - install run.sh /app/bin/
+        # FIXME: These may fall out of "pack", although I can't find them in "dist" :-/
+        - install -Dm 0644 chat.delta.desktop.desktop  /app/share/applications/chat.delta.desktop.desktop
+        - install -Dm 0644 static/chat.delta.desktop.appdata.xml  /app/share/appinfo/chat.delta.desktop.appdata.xml
+        # FIXME: Where do we find the icon?!
+        #- install -Dm 0644 /app/share/icons/HighContrast/scalable/apps/accessories-calculator.svg /app/share/icons/hicolor/scalable/apps/chat.delta.desktop.svg
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/256x256/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/128x128/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/64x64/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/32x32/apps/chat.delta.desktop.png
+        - install -Dm 0644 chat.delta.desktop.png /app/share/icons/hicolor/16x16/apps/chat.delta.desktop.png
```

Comment:
Why do you install the same 256x256 icon multiple times?

Reply:
I think I have seen these icons being installed by the binary package, but I'm unable to get them built. It's also weird, because AFAICS, the source tree only contains a 240x240 image.
@ralphtheninja do you know about the icon, e.g. how they are built and whether multiple resolutions are available? Maybe even a scalable one?

Reply:
`electron-builder` uses https://github.com/deltachat/deltachat-desktop/blob/master/build/icon.icns to build out a set of images in different resolutions

Reply:
how do we get the final icons out of that?
Is there a normal format of the icon somewhere? Like a SVG?

Reply:
@muelli If you run `npm run dist` it will build out those icons in `./dist/.icon-set`:

![screenshot from 2018-12-22 15-13-12](https://user-images.githubusercontent.com/308049/50375358-39bb8980-05fc-11e9-87ce-fe2401b2d106.png)



> Is there a normal format of the icon somewhere? Like a SVG?

Currently in `desktop` it's the `icon.icns` file mentioned above that's the canonical file, but it's obviously generated from some other material. Lets ask around and see if we can resolve this without relying on `electron-builder`.

@muelli Counter question: What format/resolution do you want/need?

@karissa How is the `icon.icns` file generated and out of what resource was it made?

@r10s Do we have any svg files of the delta chat logo?

Reply:
@ralphtheninja the logo as svg is here: https://github.com/deltachat/interface/blob/master/icons/delta-v7-pathed.svg

Reply:
> nice! When is it best to run this command? [Right now](https://github.com/flathub/flathub/pull/772/files#diff-5a918bacb13ca69f745f64af712b3e60R59), `npm install` is followed by `npm run build`, and then `pack`. Should the `dist` come before or after that?

* `npm install` always has to run first, to get all dependencies
* `npm run build` builds `static/bundle.js` which is needed before `pack`
* `npm run pack` is a lightweight version of `npm run dist` (it doesn't generate the icons and doesn't generate `.Appimage` nor `.deb` file

![screenshot from 2018-12-22 17-44-47](https://user-images.githubusercontent.com/308049/50376715-59a97800-0611-11e9-93f3-ca4d76269121.png)

> Does it build the desktop file and the appstream file, too?

Remind me, what are the names of these files?

Reply:
Ok, so the appstream file is `chat.delta.desktop.yml`? This file was provided by you iirc.

Reply:
> Remind me, what are the names of these files?

ah. Electron-builder does not provide this file, it seems: https://github.com/electron-userland/electron-builder/issues/1993

Reply:
> `npm run pack` is a lightweight version of `npm run dist` (it doesn't generate the icons and doesn't generate `.Appimage` nor `.deb` file)

Hrm. It seems unnecessary to build the .deb and AppImage file. With the SVG we should be fine, though.  It is being used now. Thanks.